### PR TITLE
Add the support of multiple root port for q35

### PR DESCRIPTION
**Pull request by jingzhao84**
Signed-off-by: Jing Zhao <jinzhao@redhat.com>
ID: 1408783

### CODE REVIEW

File: virttest/qemu_devices/qcontainer.py [hunk 1]
```diff
@@ -1748,8 +1748,11 @@ def pcic_by_params(self, name, params):
                   and you are responsible for creating the downstream ports.
         """
         driver = params.get('type', 'ioh3420')
+        pcic_params = {"id": name}
         if driver in ('ioh3420', 'x3130-upstream', 'x3130'):
             bus_type = 'PCIE'
+            if driver == "ioh3420":
```
Comment (vivianQizhu) [inline virttest/qemu_devices/qcontainer.py]: This 'if' should be independent with above 'if', can move out from the block.

Comment (vivianQizhu) [inline virttest/qemu_devices/qcontainer.py]: This 'if' should be independent with above 'if', can move out from the block.

**Comment by CongLi** [comment]
@jingzhao84 

Please remove id info from your commit log which is internal only.

Thanks.

File: virttest/qemu_devices/qcontainer.py (hunk 1 above)
Comment (jingzhao84) [inline virttest/qemu_devices/qcontainer.py]: Thanks your comments, moved the 'if' out

**Comment by jingzhao84** [comment]
@CongLi  updated the commit log and thanks your comments

**Comment by huangyum** [comment]
LGTM
Acked-by: Yumei Huang yuhuang@redhat.com

File: virttest/qemu_devices/qcontainer.py [hunk 2]
```diff
@@ -1748,10 +1748,13 @@ def pcic_by_params(self, name, params):
                   and you are responsible for creating the downstream ports.
         """
         driver = params.get('type', 'ioh3420')
+        pcic_params = {"id": name}
         if driver in ('ioh3420', 'x3130-upstream', 'x3130'):
             bus_type = 'PCIE'
```
Comment (xutian) [inline virttest/qemu_devices/qcontainer.py]:
```
if drive == "ioh3420" and params.get("slot"):
    pcic_params["slot"] = params["slot"]
```

Comment (jingzhao84) [inline virttest/qemu_devices/qcontainer.py]: Thanks your comments and replaced the following code with the above code 

+        if driver == "ioh3420" :
 +            pcic_params["slot"] = params.get("slot")

**Comment by xutian** [comment]
@jingzhao84  please update your commit summary like:

add slot option support for ioh3420 device

**Comment by jingzhao84** [comment]
@xutian update the commit log, thanks

**Comment by xutian** [comment]
LGTM

**Pull request merged**